mlir/lib/Dialect/SparseTensor/IR/SparseTensorDialect.cpp

```diff
@@ -1780,6 +1781,26 @@ void SparseTensorDialect::initialize() {
 #define GET_OP_LIST
 #include "mlir/Dialect/SparseTensor/IR/SparseTensorOps.cpp.inc"
       >();
+  declarePromisedInterface<ConcatenateOp,
+                           bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<ConvertOp, bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<LoadOp, bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<NewOp, bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<InsertOp, bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<NumberOfEntriesOp,
+                           bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<AssembleOp,
+                           bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<DisassembleOp,
+                           bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<ToCoordinatesBufferOp,
+                           bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<ToCoordinatesOp,
+                           bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<ToPositionsOp,
+                           bufferization::BufferizableOpInterface>();
+  declarePromisedInterface<ToValuesOp,
+                           bufferization::BufferizableOpInterface>();
```

Comment:
Looking at these lists, is there a chance to somehow tweak `declarePromisedInterface` so it takes a list of ops? Either by swapping the order of parameters or by doing some tuple indexing to slice the last template parameter?

Reply:
I am fine with the change for sparse, so lgtm, but can you give me some background on how this improves things? At a glance, it just looks like adding boilerplate for something that was already working (with the "crash" if not declared change only).

Reply:
It's making the system safer, it's actually a mistake that we allowed to inject interface separately without the promises: it is a hole in the system that we haven't been able to close just yet.

Reply:
> is there a chance to somehow tweak declarePromisedInterface so it takes a list of ops? 

[Declare multiple promised interfaces at once](https://github.com/llvm/llvm-project/pull/78368/commits/5f448a14789a822da251ec0b8a161507e462d7dd) should resolve this.